virttest/utils_misc.py

```diff
@@ -2844,44 +2853,42 @@ def format_linux_disk(session, did, all_disks_did, partition=False,
     :param fstype: filesystem type for the disk.
     :return Boolean: disk usable or not.
     """
-    disks = list_linux_guest_disks(session, partition)
-    logging.debug("Disks detail: %s" % disks)
-    for line in disks:
-        kname = line.split('/')[2]
-        did_list = all_disks_did[kname]
-        if did not in did_list:
-            # Continue to search target disk
+    if not all_disks_did:
+        all_disks_did = get_linux_disks_did(session).values()
```

Comment:
Should use get_linux_disks_did(session, parted).values(), otherwise parted is not used.